### PR TITLE
Test application for VK_KHR_16bit_storage

### PR DESCRIPTION
**Pull request by apazylbe**
Based on the cube sample, adds another uniform buffer with uint16_t
values.

### CODE REVIEW

**Review by AWoloszyn — APPROVED** [review]

File: application_sandbox/shader_16bit_storage/main.cpp [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+ private:
+  struct CameraData {
+    Mat44 projection_matrix;
+  };
+
+  struct ModelData {
+    Mat44 transform;
+  };
+
+  struct ColourData {
```
Comment (AWoloszyn) [inline application_sandbox/shader_16bit_storage/main.cpp:342]: We use both color and colour in the sample. We should be consistent.

**Review by AWoloszyn — CHANGES_REQUESTED** [review]
See previous comment.

File: application_sandbox/shader_16bit_storage/main.cpp [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+        VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER,  // descriptorType
+        1,                                  // descriptorCount
+        VK_SHADER_STAGE_VERTEX_BIT,         // stageFlags
+        nullptr                             // pImmutableSamplers
+    };
+    cube_descriptor_set_layouts_[2] = {
+        2,                                  // binding
+        VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER,  // descriptorType
+        1,                                  // descriptorCount
+        VK_SHADER_STAGE_VERTEX_BIT,         // stageFlags
```
Comment (krmoule) [inline application_sandbox/shader_16bit_storage/main.cpp:112]: Isn't this one going VK_SHADER_STAGE_FRAGMENT_BIT?

File: application_sandbox/shader_16bit_storage/main.cpp (hunk 1 above)
Comment (apazylbe) [reply application_sandbox/shader_16bit_storage/main.cpp]: Done, thanks!

File: application_sandbox/shader_16bit_storage/main.cpp (hunk 2 above)
Comment (apazylbe) [reply application_sandbox/shader_16bit_storage/main.cpp]: Thanks for the catch! I was using it in vertex shader at first and then forgot to change.

**Review by krmoule — APPROVED** [review]

**Review by AWoloszyn — APPROVED** [review]

**Pull request merged**